analyses/molecular-subtyping-LGAT/02-subset-fusion-files-LGAT.Rmd

```diff
@@ -0,0 +1,152 @@
+---
+title: "Annotate Fusion defining subtype status for LGAT biospecimens"
+output: html_notebook
+author: K S Gaonkar 
+date: 2020
+---
+
+As per [issue](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/790) we will be subtyping LGAT based on fusion  in the following genes:
+
+ - LGG, KIAA1549-BRAF
+contains KIAA1549-BRAF fusion
+
+ - LGG, other MAPK
+contains non-canonical BRAF fusion other than KIAA1549-BRAF
+contains RAF1 fusion
+
+- LGG, RTK
+harbors a fusion in ALK, ROS1, NTRK1, NTRK2, or NTRK3 or
+harbors a PDGFRA fusion
+
+- LGG, FGFR
+harbors FGFR1-TACC1 fusion 
+harbors FGFR1 or FGFR2 fusions
+
+- LGG, MYB/MYBL1
+harbors either a MYB-QKI fusion or other MYB or MYBL1 fusion
+
+
+### Setup
+```{r}
+library(tidyverse)
+
+# Look for git root folder
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+# get subset folder
+subset_dir <- file.path(root_dir, 
+                        "analyses",
+                        "molecular-subtyping-LGAT",
+                        "lgat-subset")
+
+# create if doesn't exist
+if (!dir.exists(subset_dir)) {
+  dir.create(subset_dir)
+}
+```
+
+### Input
+
+`lgat_metadata.tsv` from 00-LGAT-select-pathology-dx that is used for the pathology diagnosis was used to subset clinical file.
+
+`fusion_summary_lgat_foi.tsv` was generated by `fusion-summary` module for LGAT RNA biospecimens.
+
+All fusions in the above gene of interest per subtype was summarized "1" for present and "0" if absent.
+
+For kinase fusions, the following conditions needed to be satisfied for LGAT as per [PR](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/830):
+- Added all 3' kinase fusions which are in-frame and retain the kinase domain
+- For 5' kinase fusions, added those which are in-frame and retain the kinase domain
+- For 5' kinase fusions that don't meet 3., check whether it has a reciprocal fusion and whether that reciprocal is in-frame and the kinase domain is retained. Then, add those 5' kinase fusions.
+
+```{r}
+
+lgat_specimens_df<-read_tsv(file.path(subset_dir, "lgat_metadata.tsv"))
+
+# Filter to RNA-Seq samples
+lgat_rna_df <- lgat_specimens_df %>%
+  dplyr::filter(experimental_strategy == "RNA-Seq") %>%
+  dplyr::select(Kids_First_Biospecimen_ID,sample_id)
+
+# get putative oncogene fusion list
+putativeFusion <- readr::read_tsv(file.path(root_dir,
+                                            "analyses",
+                                            "fusion-summary",
+                                            "results",
+                                            "fusion_summary_lgat_foi.tsv")) %>%
```

Comment:
This will have to change when #849 goes in, to use the file from the data release. Will you create a ticket for this later change?

Reply:
We would want to use the relative path to fusion_summary_lgat_foi.tsv because those modules will be run with base histology which will then be most updated files to be used in molecular-subtyping-LGAT, right?

Reply:
Did you come to an agreement?

Reply:
We agreed to use relative paths in subtyping modules to get the most updated files from fusion-summary, right @jharenza?

Reply:
yes, relative